drivers/media/i2c/irs1125.c

```diff
@@ -0,0 +1,1137 @@
+    int ret;
+    struct v4l2_subdev *sd;
+    struct device_node *np = client->dev.of_node;
+    u32 xclk_freq;
+    int gpio_num;
+
+    sensor = devm_kzalloc(dev, sizeof(*sensor), GFP_KERNEL);
+    if (!sensor)
+        return -ENOMEM;
+
+    if (IS_ENABLED(CONFIG_OF) && np) {
+        ret = irs1125_parse_dt(np, sensor);
+        if (ret) {
+            dev_err(dev, "DT parsing error: %d\n", ret);
+            return ret;
+        }
+    }
+
+    /* get system clock (xclk) */
+    sensor->xclk = devm_clk_get(dev, NULL);
+    if (IS_ERR(sensor->xclk)) {
+        dev_err(dev, "could not get xclk");
+        return PTR_ERR(sensor->xclk);
+    }
+
+    xclk_freq = clk_get_rate(sensor->xclk);
+    if (xclk_freq != 25000000) {
+        dev_err(dev, "Unsupported clock frequency: %u\n", xclk_freq);
+        return -EINVAL;
+    }
+    
+    sensor->num_seq = 5;
+    
+    /* Request the power down GPIO asserted */
+    sensor->reset = devm_gpiod_get(&client->dev, "pwdn",
+                           GPIOD_OUT_LOW);
+    
+    if (IS_ERR(sensor->reset)) {
+        dev_err(dev, "could not get reset");
+        return PTR_ERR(sensor->reset);
+    }
+    
+    
+    gpio_num = desc_to_gpio(sensor->reset);
+
+    mutex_init(&sensor->lock);
+
+    sd = &sensor->sd;
+    v4l2_i2c_subdev_init(sd, client, &irs1125_subdev_ops);
+    
+    ret = irs1125_ctrls_init(sensor, dev);
+    if (ret < 0)
+        goto error;
+
+    sensor->sd.internal_ops = &irs1125_subdev_internal_ops;
+    sensor->sd.flags |= V4L2_SUBDEV_FL_HAS_DEVNODE;
+
+    sensor->pad.flags = MEDIA_PAD_FL_SOURCE;
+    sd->entity.function = MEDIA_ENT_F_CAM_SENSOR;
+    ret = media_entity_pads_init(&sd->entity, 1, &sensor->pad);
+    if (ret < 0)
+        goto mutex_remove;
+
+    gpiod_set_value_cansleep(sensor->reset, 1);
+    msleep(RESET_ACTIVE_DELAY_MS);
+    
+    ret = irs1125_detect(sd);
+    if (ret < 0)
+        goto error;
+
+    ret = irs1125_ident_setup(sensor, dev);
+    if (ret < 0)
+        goto error;
+
+    gpiod_set_value_cansleep(sensor->reset, 0);
+    
+    ret = v4l2_async_register_subdev(sd);
+    if (ret < 0)
+        goto error;
+
+    dev_dbg(dev, "Infineon IRS1125 camera driver probed\n");
+    return 0;
+error:
+    media_entity_cleanup(&sd->entity);
+mutex_remove:
+    mutex_destroy(&sensor->lock);
+    return ret;
+}
+
+static int irs1125_remove(struct i2c_client *client)
+{
+    struct v4l2_subdev *sd = i2c_get_clientdata(client);
+    struct irs1125 *irs1125 = to_state(sd);
+
+    v4l2_async_unregister_subdev(&irs1125->sd);
+    media_entity_cleanup(&irs1125->sd.entity);
+    v4l2_device_unregister_subdev(sd);
+    mutex_destroy(&irs1125->lock);
+    v4l2_ctrl_handler_free(&irs1125->ctrl_handler);
+
+    return 0;
+}
+
+static const struct i2c_device_id irs1125_id[] = {
+    { "irs1125", 0 },
+    { }
+};
+MODULE_DEVICE_TABLE(i2c, irs1125_id);
+
+#if IS_ENABLED(CONFIG_OF)
+static const struct of_device_id irs1125_of_match[] = {
+    { .compatible = "irs1125" },
```

Comment:
The compatible string should always be "vendor,model". eg "ovti,ov5647,ov5647", or "sony,imx219".
As defined in https://github.com/raspberrypi/linux/blob/rpi-4.19.y/Documentation/devicetree/bindings/vendor-prefixes.txt#L175, Infineon Technologies is "infineon"

Reply:
Thanks for the hint. I head no idea about the vendor,model string. Fixed now